acme_diags/driver/tc_analysis_driver.py

```diff
@@ -0,0 +1,541 @@
+    for i in range(0, len(mon)):
+        k = int(mon[i])
+        seasonal_cycle[k - 1] = seasonal_cycle[k - 1] + 1
+
+    return seasonal_cycle / np.sum(seasonal_cycle)
+
+
+def run_diag(parameter: "CoreParameter") -> "CoreParameter":
+    """Runs the tropical cyclone analysis diagnostic.
+
+    :param parameter: Parameters for the run
+    :type parameter: CoreParameter
+    :raises ValueError: [description]
+    :return: Parameters for the run
+    :rtype: CoreParameter
+    """
+    test_data_path = parameter.test_data_path
+    reference_data_path = parameter.reference_data_path
+    test_name = parameter.test_name
+    run_type = parameter.run_type
+    test_start_yr = parameter.test_start_yr
+    test_end_yr = parameter.test_end_yr
+
+    # Use the basin location info from [Balaguru et al. 2020]
+    # obs for TC frequency of each basins
+
+    # https://www.jstage.jst.go.jp/article/jmsj/84/2/84_2_259/_pdf/-char/ja
+
+    # ns_obs = 26.7 + 10.4 + 18.1 + 8.6 + 4.6 + 15.4
+    # pdf_wp_obs = 26.7 / ns_obs
+    # pdf_sp_obs = 10.4 / ns_obs
+    # pdf_ep_obs = 18.1 / ns_obs
+    # pdf_at_obs = 8.6 / ns_obs
+    # pdf_ni_obs = 4.6 / ns_obs
+    # pdf_si_obs = 15.4 / ns_obs
+
+    test_te_file = os.path.join(
+        test_data_path,
+        "cyclones_stitch_{}_{}_{}.dat".format(test_name, test_start_yr, test_end_yr),
+    )
+    test_cyclones_file = os.path.join(
+        test_data_path,
+        "cyclones_hist_{}_{}_{}.nc".format(test_name, test_start_yr, test_end_yr),
+    )
+    test_cyclones_hist = cdms2.open(test_cyclones_file)(
+        "density", lat=(-60, 60, "ccb"), squeeze=1
+    )
+    test_aew_file = os.path.join(
+        test_data_path,
+        "aew_hist_{}_{}_{}.nc".format(test_name, test_start_yr, test_end_yr),
+    )
+    test_aew_hist = cdms2.open(test_aew_file)(
+        "density", lat=(0, 35, "ccb"), lon=(-180, 0, "ccb"), squeeze=1
+    )
+
+    test_data = collections.OrderedDict()
+    ref_data = collections.OrderedDict()
+
+    test_data["metrics"] = generate_tc_metrics_from_te_stitch_files(test_te_file)
+    test_data["cyclone_density"] = test_cyclones_hist
+    test_data["aew_density"] = test_aew_hist
+    test_num_years = int(test_end_yr) - int(test_start_yr) + 1
+    test_data["aew_num_years"] = test_num_years  # type: ignore
+    test_data["cyclone_num_years"] = test_num_years  # type: ignore
+    parameter.test_title = "{} ({}-{})".format(
+        parameter.test_name, test_start_yr, test_end_yr
+    )
+
+    if run_type == "model_vs_model":
+        ref_name = parameter.ref_name
+        ref_start_yr = parameter.ref_start_yr
+        ref_end_yr = parameter.ref_end_yr
+        ref_te_file = os.path.join(
+            reference_data_path,
+            "cyclones_stitch_{}_{}_{}.dat".format(ref_name, ref_start_yr, ref_end_yr),
+        )
+        ref_cyclones_file = os.path.join(
+            reference_data_path,
+            "cyclones_hist_{}_{}_{}.nc".format(test_name, test_start_yr, test_end_yr),
```

Comment:
should be `(ref_name, ref_start_yr, ref_end_yr)`, I missed this earlier.

Reply:
Fixed